lib/stdlib/src/binary.erl

```diff
@@ -363,6 +362,85 @@ get_opts_replace([{insert_replaced,N} | T],{Part,Global,_Insert}) ->
 get_opts_replace(_,_) ->
     throw(badopt).
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Hex encoding functions
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-spec to_hex(Bin) -> Bin2 when
+      Bin :: binary(),
+      Bin2 :: binary().
+to_hex(Bin) when is_binary(Bin) ->
+    to_hex(Bin, [lowercase]);
+to_hex(Bin) ->
+    badarg_with_info([Bin]).
+
+-spec to_hex(Bin, Options) -> Bin2 when
+      Bin :: binary(),
+      Options :: [Option],
+      Option :: lowercase | uppercase,
+      Bin2 :: binary().
+to_hex(Bin, Options) when is_binary(Bin), is_list(Options) ->
+    case proplists:get_bool(uppercase, Options) of
+        false ->
+            to_hex(Bin, fun enc_hex_digit_lower/1, <<>>);
+        true ->
+            to_hex(Bin, fun enc_hex_digit_upper/1, <<>>)
+    end;
+to_hex(Bin, Options) ->
+    badarg_with_info([Bin, Options]).
+
+-spec to_hex(Bin, Enc16, Acc) -> Bin2 when
+      Bin :: binary(),
+      Enc16 :: fun((0..15) -> byte()),
+      Acc :: binary(),
+      Bin2 :: Acc.
+to_hex(<<>>, _Enc16, Acc) ->
+    Acc;
+to_hex(<<A0:4, B0:4, Rest/binary>>, Enc16, Acc) ->
+    A = Enc16(A0),
+    B = Enc16(B0),
+    to_hex(Rest, Enc16, <<Acc/binary, A, B>>).
+
+-spec enc_hex_digit_upper(0..15) -> byte().
+enc_hex_digit_upper(Char) when Char =< 9 ->
+    Char + $0;
+enc_hex_digit_upper(Char) when Char =< 15 ->
+    Char + $A - 10.
+
+-spec enc_hex_digit_lower(0..15) -> byte().
+enc_hex_digit_lower(Char) when Char =< 9 ->
+    Char + $0;
+enc_hex_digit_lower(Char) when Char =< 15 ->
+    Char + $a - 10.
+
+-spec from_hex(Bin) -> Bin2 when
+      Bin :: binary(),
+      Bin2 :: binary().
+from_hex(Bin) when is_binary(Bin) ->
+    from_hex(Bin, <<>>);
+from_hex(Bin) ->
+    badarg_with_info([Bin]).
+
+-spec from_hex(Bin, Acc) -> Bin2 when
+      Bin :: binary(),
+      Acc :: binary(),
+      Bin2 :: Acc.
+from_hex(<<>>, Acc) ->
+    Acc;
+from_hex(<<A0:8, B0:8, Rest/binary>>, Acc) ->
+    A = dec_hex_char(A0),
+    B = dec_hex_char(B0),
+    from_hex(Rest, <<Acc/binary, A:4, B:4>>).
+
```

Comment:
The code does not handle the case when there is an odd number of characters, it will just crash with function_clause.

Reply:
Nice catch! Fixed.